src/engine/Actor.ts

```diff
@@ -426,6 +426,27 @@ module ex {
        // Override me
     }
 
+    /**
+     * Gets wether the actor is Initialized 
+     */
+    public get isInitialized(): boolean {
+       return this._isInitialized;
+    }
+
+    /**
+     * Initializes this actor and all it's child actors, meant to be called by the Scene before first update.
+     */
+    public initialize(engine: Engine) {
```

Comment:
This is a bigger discussion, but I think we'll need to make a decision soon about whether or not we should tell people to override these methods but end up leaving a landmine that blows up if you forget to call `super.whatever`. It's happened with draw, update, and now initialize.

I think we should consider moving all lifecycle events to internal methods and using the `onWhatever` convention to denote methods that can safely be overridden without any side-effects.